### ISSUE TITLE
Disable `no-missing-eof-newline` for HTML-linting(?)

### ISSUE THREAD

**Issue by ndarville** [stylelint/stylelint#1226]
**tl;dr:** `-e` looks for `no-missing-eof-newline`, which makes sense for CSS files, but not `<style>`.

* * * *

> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Inadequate rule that makes sense for (S)CSS-linting, but not for `<style>`-linting.

> Which rule, if any, is this issue related to?

`no-missing-eof-newline`.

> What stylelint configuration is needed to reproduce this issue?

<https://github.com/ndarville/style/blob/master/css/.stylelintrc>, although I think it’s a general problem.

> Which version of stylelint are you using?

`6.3.3`.

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `stylelint -e _site/{asides,diverse,projects}/*.html _site/{asides,diverse,projects}/*/*.html`.

> What actually happened (e.g. what warnings or errors you are getting)?

```html
<style type=“text/css”>a strong { font-weight: normal; }</style>
```

```shell
_site/asides/danish-handbook/index.html
 1:33  ✖  Unexpected missing newline at end of file   no-missing-eof-newline
```

## Options ##

* Support rule exceptions in CLI (`—ignore no-missing-eof-newline`)
* Support rule exception in config
* Hardcode manual exception for `no-missing-eof-newline`

<3

**Comment by davidtheclark** [comment]
What if we just had the code that parses `<style>` tags always add a newline at the end of the content? @blake-newman do you see any problem with that?

**Comment by ndarville** [comment]
I’m down with that.

**Comment by evilebottnawi** [comment]
@davidtheclark i think it is good solution

**Comment by blake-newman** [comment]
@davidtheclark Seems like a valid solution. Only issue with `<style type=“text/css”>a strong { font-weight: normal; }</style>`

Is that it won't map correctly, however this is also similar issue with eslint, and is yet to be resolved.

**Comment by evilebottnawi** [comment]
@blake-newman that about a problem specifically refers?

**Comment by evilebottnawi** [comment]
@jeddy3 i think after rename rule, what happens is that we expect, we also took out the ability to extraction from HTML outside core, i think we can close this issue. It is belongs to plugin, which extract HTML.

**Comment by jeddy3** [comment]
Yep.
